brainwidemap/decoding/functions/process_outputs.py

```diff
@@ -0,0 +1,382 @@
+
+def get_val_from_realsession(reseidreg, value_name, RUN_ID=1):
+    '''
+    helper function meant to take a pandas dataframe, reseidreg, of decoding results from
+    a single eid and a single region.  The value corresponding to the value_name column,
+    RUN_ID number, and pseudo_id=-1 is returned.  There should only be one such value, so 
+    if the dataframe contains mutliple, then None is returned.  If the value is None, then
+    None is returned.       
+
+    Parameters
+    ----------
+    reseidreg : pandas dataframe
+        Decoding results from a single eid and single region. 
+        .
+    value_name : str
+        the name of a column in reseidreg
+        .
+    RUN_ID=1 : int
+        The run id of the desired decoding value.  Decoding is often repeated multiple 
+        times to reduce variability, and run id indexes these repetitions. Default is 1.
+
+    Returns
+    -------
+    numpy array with a single value
+
+    '''
+
+    my_vals = list(reseidreg.loc[(reseidreg['pseudo_id']==-1)&(reseidreg['run_id']==RUN_ID), value_name])
+    if (len(my_vals) != 1) or (my_vals[0] is None):
+        return None
+    return np.array(my_vals[0])
+
+def check_scores(my_preds, my_targets, score_name, real_scores):
+    '''
+    checks whether the predictions produce the same performance 
+    scores as those that are given in the real_scores array.  
+    Compares predictions (my_preds) and targets (my_targets) to produce the
+    desired score_name and returns True if calculated scores all match the
+    recored scores
+
+    Parameters
+    ----------
+    my_preds : list of 2-d arrays
+        2-d arrays are decoder predictions across trials.  Assumed that 
+        second dimension has size 1 i.e. not wheel decoding.  The list indexes
+        across run ids
+    my_targets : 2-d array
+        Same format as my_preds, but not a list (targets are the same for all run ids).
+    score_name : str, 'balanced_acc_test' or 'R2_test'
+        These are the two test statistics used to quantify BWM decoding performance: 
+        balanced_accuracy_score and r2_score from sklearn.metrics. 
+    real_scores : 1-d array
+        The recored scores of decoding performance across run id.
+
+    Returns
+    -------
+    boolean
+
+    '''
+
+    my_targets_flat = my_targets[:,0]
+    my_preds_flat = [my_preds[pi][:,0] for pi in range(my_preds.shape[0])]
+    assert len(my_targets_flat.shape)==1
+    assert np.all(np.array([len(p.shape) for p in my_preds_flat])==1)
+    
+    if score_name == 'balanced_acc_test':
+        calc_score = lambda x: balanced_accuracy_score(my_targets_flat, x)
+    elif score_name == 'R2_test':
+        calc_score = lambda x: r2_score(my_targets_flat, x)
+    my_calc_real_scores = [calc_score(p) for p in my_preds_flat]
+    isequal_scores = [my_calc_real_scores[i]==real_scores[i] for i in range(len(real_scores))]
+    return np.all(np.array(isequal_scores))
+        
+
+def create_pdtable_from_raw(res, 
```

Comment:
Despite the docstring I am finding it hard to review this function. It would really help to have more comments to explain what happens at each step. For now I am not reviewing it in more depth